chore(#921): Clarify the release process (#1012)
medic/cht-docs: content/en/contribute/code/releasing.md

## CHT Core

> A release is a set of changes that is planned to be used by at least one implementation that uses the CHT.

Reviewer: This definition seems to say we won't make a release _unless_ we have a project planning to use the new version. That seems ideal, but is planning an upgrade with a project a prerequisite to release a bugfix?

Here's a possible suggestion, but may need further wordsmithing:
```suggestion
A release is a set of code changes bundled together, ideally with at least one deployment of CHT apps ready to make use of it.
```

### Building & Releasing CHT Core Changes

The high-level steps for a release are as follows:

> * The team sees an opportunity they want to go after. The opportunity addresses a need of at least one implementation that uses the CHT and will be used by that implementation after the release.

Reviewer: Who is "the team" here? Is it a Focused Working Group, or Medic's Product team? Given that this is the first use of "the team" on this page, worth being clearer about it.

Author: By team, I meant FWG. Just clarified it, thanks.

> * The Focused Working Group agrees on a solution for it.

Reviewer 2: This content has a lot overlap with what we talk about on the [Feature Release page](https://docs.communityhealthtoolkit.org/core/releases/feature_releases/). Maybe we should consider merging the two pages or possibly moving the FR page to `content/en/contribute/code/*` so it's side by side?

Author: Great point @mrjones-plip, thanks for redirecting me to the FR page. I moved it under the `content/en/contribute/code/*` as it makes sense to keep it together with the release guidelines.

* Tickets are added to the [Product Team Activities board](https://github.com/orgs/medic/projects/134/views/3) for what's being built.
* A [release manager](#release-manager) is assigned from the team.
* The release manager [creates an issue](https://github.com/medic/cht-core/issues/new/choose) for either a [Major/Minor or Patch](#major-minor-patch-release) release issue template and follows the process.

> * Change is built by a developer together with [quality assistance]({{% ref "/contribute/medic/product-development-process/quality-assistance" %}}).

Reviewer 3: Did you mean "change"? I'm not sure, but I think you may have meant code or feature or something different.

```suggestion
* Code is built by a developer together with [quality assistance]({{% ref "/contribute/medic/product-development-process/quality-assistance" %}}).
```

Author: Yes, code is better and consistent with the rest of the steps. Thank you

* [Code is reviewed]({{% ref "/contribute/code/workflow#code-reviews" %}}).
* Code is merged.

> * Code is released.

Reviewer 3: I read this page and this section in particular to be instructions on what the release process is. With that context though, this line would seem to be a recursive reference. More likely though I have misunderstood the intention of the page & section. How about if the section title is "Building & Releasing Software"?

Author: Yes, `code` is better and consistent with the rest of the steps. Thank you

Author: How about `Building & Releasing CHT Core Changes`?

### Release Manager
The overall coordination and operation of the release process are the responsibility of the release manager.

> The release manager must perform several tasks for new release, such as coordinating with team members and following all the steps in the [release issue process](https://github.com/medic/cht-core/issues/new/choose), some of them being manual. The release manager must have adequate permissions to the repositories where the release is made.

Reviewer 3: ```suggestion
The release manager must perform several tasks for a new release, such as coordinating with team members and following all the steps in the [release issue process](https://github.com/medic/cht-core/issues/new/choose), some of them being manual. The release manager must have adequate permissions to the repositories where the release is made.
```

### Major/Minor/Patch Release
{{% alert title="Note" %}} The following classification is defined by the [Semantic Versioning 2.0.0](https://semver.org). {{% /alert %}}

Given a version number `MAJOR.MINOR.PATCH`, increment the:

> * `MAJOR` version when the release adds incompatible API changes

Reviewer: > incompatible API changes

Specifying "API" could be seen as relatively narrow. Is there a broader definition in our context? For instance, "when the release has changes that require manual intervention for your apps to work."

> * `MAJOR` version the release adds incompatible API changes
> * `MINOR` version the release adds functionality in a backward-compatible manner
> * `PATCH` version the release adds backward-compatible bug fixes

Reviewer 3: These lines read a bit funny. They could be put into a table form, or just inserting a "when" as the [referenced page](https://semver.org/#summary) does is good:
```suggestion
* `MAJOR` version when the release adds incompatible API changes
* `MINOR` version when the release adds functionality in a backward-compatible manner
* `PATCH` version when the release adds backward-compatible bug fixes
```

Author: 😆 nice one
